--- a/SSL.pm	2010-08-20 21:10:06.000000000 -0400
+++ b/SSL.pm	2016-04-11 11:48:55.871008484 -0400
@@ -1353,6 +1353,8 @@
 		$ctx = /^sslv2$/i ? Net::SSLeay::CTX_v2_new()	 :
 			   /^sslv3$/i ? Net::SSLeay::CTX_v3_new()	 :
 			   /^tlsv1$/i ? Net::SSLeay::CTX_tlsv1_new() :
+			   /^tlsv1_?1$/i ? Net::SSLeay::CTX_tlsv1_1_new() :
+			   /^tlsv1_?2$/i ? Net::SSLeay::CTX_tlsv1_2_new() :
 							Net::SSLeay::CTX_new();
 	}
 
@@ -1656,9 +1658,13 @@
 
 =item SSL_version
 
-Sets the version of the SSL protocol used to transmit data.	 The default is SSLv2/3,
-which auto-negotiates between SSLv2 and SSLv3.	You may specify 'SSLv2', 'SSLv3', or
-'TLSv1' (case-insensitive) if you do not want this behavior.
+Sets the version of the SSL protocol used to transmit data.
+'SSLv23' uses a handshake compatible with SSL2.0, SSL3.0, and TLS1.x, while
+'SSLv2', 'SSLv3', 'TLSv1', 'TLSv1_1', or 'TLSv1_2' restrict handshake and
+protocol to the specified version.
+All values are case-insensitive.  Instead of 'TLSv1_1' and 'TLSv1_2' one can
+also use 'TLSv11' and 'TLSv12'.  Support for 'TLSv1_1' and 'TLSv1_2' requires
+recent versions of Net::SSLeay and openssl.
 
 =item SSL_cipher_list
 
